src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1137,6 +1144,83 @@ public ContainerStats stats(final String containerId)
     }
   }
 
+  @Override
+  public List<Network> listNetworks() throws DockerException, InterruptedException {
+    final WebTarget resource = resource().path("networks");
+    return request(GET, NETWORK_LIST, resource, resource.request(APPLICATION_JSON_TYPE));
+  }
+
+  @Override
+  public Network inspectNetwork(String networkId) throws DockerException, InterruptedException {
+    final WebTarget resource = resource().path("networks").path(networkId);
+    try {
+      return request(GET, Network.class, resource, resource.request(APPLICATION_JSON_TYPE));
+    } catch (DockerRequestException e) {
+      switch (e.status()) {
+        case 404:
+          throw new NetworkNotFoundException(networkId, e);
+        default:
+          throw e;
+      }
+    }
+  }
+
+  @Override
+  public NetworkCreation createNetwork(NetworkConfig networkConfig)
+      throws DockerException, InterruptedException {
+    final WebTarget resource = resource().path("networks").path("create");
+
+    return request(POST, NetworkCreation.class, resource, resource.request(APPLICATION_JSON_TYPE),
+                   Entity.json(networkConfig));
+  }
+
+  @Override
+  public void removeNetwork(String networkId) throws DockerException, InterruptedException {
+    try {
+      final WebTarget resource = resource().path("networks").path(networkId);
+      request(DELETE, resource, resource.request(APPLICATION_JSON_TYPE));
+    } catch (DockerRequestException e) {
+      switch (e.status()) {
+        case 404:
+          throw new NetworkNotFoundException(networkId, e);
+        default:
+          throw e;
+      }
+    }
+  }
+
+  @Override
+  public void connectToNetwork(String containerId, String networkId)
+      throws DockerException, InterruptedException {
+    manageNetworkConnection(containerId, "connect", networkId);
+  }
+
+  @Override
+  public void disconnectFromNetwork(String containerId, String networkId)
+      throws DockerException, InterruptedException {
+    manageNetworkConnection(containerId, "disconnect", networkId);
+  }
+
+  private void manageNetworkConnection(String containerId, String methodname, String networkId)
+      throws DockerException, InterruptedException {
+    final WebTarget resource = resource().path("networks").path(networkId).path(methodname);
+
+    Map<String, String> request = new HashMap<>();
+    request.put("Container", containerId);
+    Response response =
+        request(POST, Response.class, resource, resource.request(APPLICATION_JSON_TYPE),
+                Entity.json(request));
+    switch (response.getStatus()) {
+      case 200:
+        return;
+      case 404:
+        throw new ContainerNotFoundException(containerId);
+      case 500:
+        throw new DockerException(response.readEntity(String.class));
+    }
+    System.out.println(response.getStatus());
```

Comment:
Removed

2015-12-28 18:36 GMT+01:00 David Xia <notifications@github.com>:

> In src/main/java/com/spotify/docker/client/DefaultDockerClient.java
> <https://github.com/spotify/docker-client/pull/307#discussion_r48493435>:
>
> > +    final WebTarget resource = resource().path("networks").path(networkId).path(methodname);
> > +
> > +    Map<String, String> request = new HashMap<>();
> > +    request.put("Container", containerId);
> > +    Response response =
> > +        request(POST, Response.class, resource, resource.request(APPLICATION_JSON_TYPE),
> > +                Entity.json(request));
> > +    switch (response.getStatus()) {
> > +      case 200:
> > +        return;
> > +      case 404:
> > +        throw new ContainerNotFoundException(containerId);
> > +      case 500:
> > +        throw new DockerException(response.readEntity(String.class));
> > +    }
> > +    System.out.println(response.getStatus());
>
> Do we need this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/docker-client/pull/307/files#r48493435>.
>



-- 
Damiano Pezzotti
...................................... . .
Software Architect | Alambitco
mobile +39 333 2116258
fax +39 02 700432209
email damy@alambitco.com
web www.alambitco.com
flickr www.flickr.com/photos/damianopezzotti
linkedin www.linkedin.com/in/damianopezzotti
pgp id 0x12DFD989